overlord/configstate/configcore/store.go

```diff
@@ -70,9 +70,14 @@ func handleStoreAccess(_ sysconfig.Device, cfg ConfGetter, _ *fsOnlyContext) err
 		return err
 	}
 
-	if err := os.MkdirAll(filepath.Dir(dirs.SnapRepairConfigFile), 0755); err != nil {
+	configFilePath := dirs.SnapRepairConfigFile
+	if opts != nil && opts.RootDir != "" {
+		configFilePath = dirs.SnapRepairConfigFileUnder(opts.RootDir)
+	}
```

Comment:
we want a test for this by using configcore.FilesystemOnlyApply as well and not just Run, if you look there should be examples in the preexisting test files